tests/workers/rollout/test_async_sglang_server.py

```diff
@@ -40,23 +40,70 @@ def server_config(self):
-    @pytest.mark.filterwarnings("ignore:Ray state API is no longer experimental:DeprecationWarning")
-    async def test_init_engine(self, mock_start_fastapi_server, mock_list_actors, server_config, mock_ray_actor):
+    @pytest.mark.filterwarnings("ignore:Ray state API is no longer experimental:DeprecationWarning")  # Keep existing filter
+    async def test_init_engine(self, mock_start_fastapi_server, mock_list_actors, server_config, mock_ray_actor):  # mock_ray_actor is kept for now
         mock_list_actors.return_value = [
+            {"name": "test_prefixWorkerDict_1:0", "namespace": "test"},
+            {"name": "test_prefixWorkerDict_1:1", "namespace": "test"},
             {"name": "test_prefixWorkerDict_0:0", "namespace": "test"},
             {"name": "test_prefixWorkerDict_0:1", "namespace": "test"},
             {"name": "test_prefixWorkerDict_1:2", "namespace": "test"},
+            {"name": "test_prefixWorkerDict_1:3", "namespace": "test"},
+            {"name": "test_prefixWorkerDict_0:2", "namespace": "test"},
+            {"name": "test_prefixWorkerDict_0:3", "namespace": "test"},
         ]
         from verl.workers.rollout.sglang_rollout.async_sglang_server import AsyncSglangServer
 
         ActualClassToInstantiate = AsyncSglangServer
         if hasattr(AsyncSglangServer, "__ray_metadata__") and hasattr(AsyncSglangServer.__ray_metadata__, "modified_class"):
             ActualClassToInstantiate = AsyncSglangServer.__ray_metadata__.modified_class
 
-        with patch("verl.workers.rollout.sglang_rollout.async_sglang_server.ray.get_actor", return_value=mock_ray_actor):
-            instance = ActualClassToInstantiate(server_config, 2, 0, "test_prefix")
+        def mock_get_actor_side_effect(name, namespace=None):  # Changed 'actor_name_arg' to 'name'
+            # Create a new mock actor for each call
+            actor_mock = MagicMock()
+            actor_mock.execute_method.remote = AsyncMock(return_value={"content": "mocked response"})
+            actor_mock.resume.remote = AsyncMock()
+            actor_mock.offload.remote = AsyncMock()
 
+            # Support .name attribute access
+            actor_mock.name = name  # Use 'name' here
+
+            # Support ['name'] item access by mocking __getitem__
+            def getitem_mock(key):
+                if key == "name":
+                    return name  # Use 'name' here
+                # For other keys, return a new MagicMock to mimic default behavior or raise KeyError
+                # Returning a MagicMock is consistent with the original error's cause for unmocked keys
+                return MagicMock(name=f"mock.__getitem__('{key}')")
+
+            actor_mock.__getitem__.side_effect = getitem_mock
+
+            return actor_mock
+
+        with patch("verl.workers.rollout.sglang_rollout.async_sglang_server.ray.get_actor", side_effect=mock_get_actor_side_effect):
```

Comment:
Including the comment `# Verify instance.workers is correctly populated` will make it easier to understand.